src/main/java/org/opentripplanner/updater/alerts/GtfsEnhancedRealtimeAlertsUpdater.java

```diff
@@ -1,139 +1,246 @@
+        this.updaterManager = updaterManager;
     }
 
     @Override
-    protected void configurePolling(Graph graph, JsonNode config) {
+    protected void configurePolling(Graph graph, JsonNode config) throws Exception {
+        // TODO: add options to choose different patch services
+        AlertPatchService alertPatchService = new AlertPatchServiceImpl(graph);
+        this.alertPatchService = alertPatchService;
         String url = config.path("url").asText();
         if (url == null) {
             throw new IllegalArgumentException("Missing mandatory 'url' parameter");
         }
         this.url = url;
-        LOG.info("Creating enhanced alert feed updater running every {} seconds: {}", pollingPeriodSeconds, url);
+        this.earlyStart = config.path("earlyStartSec").asInt(0);
+        this.feedId = config.path("feedId").asText();
+        if (config.path("fuzzyTripMatching").asBoolean(false)) {
+            this.fuzzyTripMatcher = new GtfsRealtimeFuzzyTripMatcher(graph.index);
+        }
+        LOG.info("Creating real-time alert updater running every {} seconds : {}", pollingPeriodSeconds, url);
     }
 
     @Override
-    public void setGraphUpdaterManager(GraphUpdaterManager updaterManager) {
-        this.updaterManager = updaterManager;
+    public void setup(Graph graph) {
+        if (updateHandler == null) {
+            updateHandler = new AlertsUpdateHandler();
+        }
+        updateHandler.setEarlyStart(earlyStart);
+        updateHandler.setFeedId(feedId);
+        updateHandler.setAlertPatchService(alertPatchService);
+        updateHandler.setFuzzyTripMatcher(fuzzyTripMatcher);
     }
 
     @Override
-    public void setup(Graph graph) {}
+    protected void runPolling() {
+        try {
+            InputStream data = HttpUtils.getData(url);
+            if (data == null) {
+                throw new RuntimeException("Failed to get data from url " + url);
+            }
 
-    @Override
-    public void teardown() {}
+            final FeedMessage feed = parseJson(IOUtils.toString(data));
 
-    public String toString() {
-        return "GtfsEnhancedRealtimeAlertsUpdater(" + url + ")";
-    }
+            long feedTimestamp = feed.getHeader().getTimestamp();
+            if (feedTimestamp <= lastTimestamp) {
+                LOG.info("Ignoring feed with an old timestamp.");
+                return;
+            }
 
-    private Map<String, List<EnhancedAlert>> parseJson(String jsonString) throws Exception {
-        Map<String, List<EnhancedAlert>> result = new HashMap<>();
+            // Handle update in graph writer runnable
+            updaterManager.execute(new GraphWriterRunnable() {
+                @Override
+                public void run(Graph graph) {
+                    updateHandler.update(feed);
+                }
+            });
+
+            lastTimestamp = feedTimestamp;
+        } catch (Exception e) {
+            LOG.error("Error reading gtfs-realtime feed from " + url, e);
```

Comment:
should this (and other logging/strings) say realtime enhanced instead?

Reply:
Fixed